src/main/java/com/example/eatmate/app/domain/chat/dto/response/ChatMessageListDto.java

```diff
@@ -0,0 +1,31 @@
+package com.example.eatmate.app.domain.chat.dto.response;
+
+import java.util.List;
+
+import org.springframework.data.domain.Slice;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@NoArgsConstructor
+public class ChatMessageListDto {
+
+	private List<ChatMessageResponseDto> chats;
+	private int pageNumber;
+	private boolean isLast;
+
+	@Builder
+	private ChatMessageListDto(Slice<ChatMessageResponseDto> chatList) {
+		this.chats = chatList.getContent();
+		this.pageNumber = chatList.getNumber();
+		this.isLast = chatList.isLast();
+	}
+
+	public static ChatMessageListDto from(Slice<ChatMessageResponseDto> chatList) {
+		return ChatMessageListDto.builder()
+			.chatList(chatList)
+			.build();
+	}
+}
```

Comment:
Builder 패턴을 사용할 때 Slice 객체를 직접 매개변수로 받기보다 각각의 필드값을 받는 것이 좋아보입니다!
추후 또다른 메소드가 필요할 때, Builder 패턴의 장점인 유연한 객체 생성을 제대로 활용할 수 있을 것 같습니다!!

```suggestion
	@Builder
	private ChatMessageListDto(List<ChatMessageResponseDto> chats, int pageNumber, boolean isLast) {
		this.chats = chats;
		this.pageNumber = pageNumber;
		this.isLast = isLast;
	}

	public static ChatMessageListDto from(Slice<ChatMessageResponseDto> chatList) {
		return ChatMessageListDto.builder()
			.chatList(chatList.getContent())
			.pageNumber(chatList.getNumber())
			.isLast(chatList.isLast())
			.build();
	}
}
```

Reply:
좋은 의견 감사합니다! 확실히 메서드에서 각각을 받고 builder 함수를 통해 만드는 것이 맞다고 생각됩니다! 확장의 측면, 빌더의 성질을 생각해보면 그 방식이 더 적절한 것 같습니다! 해당 부분도 수정하겠습니다